BankCOR/src/edu/ccsu/cs505/accounts/BankAccount.java

```diff
@@ -0,0 +1,83 @@
+package edu.ccsu.cs505.accounts;
+import edu.ccsu.cs505.insufficientfunds.*;
+
+import java.text.NumberFormat;
+
+
+ /**
+  * This class keeps track of all the accounts inside of a bank account. 
+  * 
+  * BankAccount is an extension of the Account class. 
+  * 
+  * @author josephtanasi10
+  */
+
+public class BankAccount extends Account 
+{
+
+	/**
+	 * This constructs a bank account with a balance that is inputed by the user.
+	 * 
+	 * @param balance The amount of money the user inputs into the bank account object.
+	 */
+	public BankAccount(double balance){
+		this.balance  = balance;
+	}
+	
+	/**
+	 * This class deducts the max amount of each account in the chain until the amount is satisfied.
+	 * 
+	 * @throws InsufficientFundsException when the total amount available 
+	 * across all accounts is insufficient. 
+	 * @param amount
+	 */
+	 @Override
+     public void deduct(double amount) throws InsufficientFundsException{
+
+        double originalBalance = balance;
+    	    
+        NumberFormat fmt = NumberFormat.getCurrencyInstance();
+    	  
+    	if(amount <= balance)
+    	balance -= amount;
+
+    	else
+    	{
+    	double remaining = (amount - balance);
+    	 		
+    	try
+    	{
+    	if(nextAccount == null)
+    		throw new InsufficientFundsException("Insufficient Funds!");
+    	balance = 0;
+    				
+    	nextAccount.deduct(remaining);
+
+    	}
+    	catch(InsufficientFundsException e){
+    				
+    	balance = originalBalance;
+    	     throw new InsufficientFundsException("Insufficient Funds!");
```

Comment:
Here rather than throwing a new InsufficientFundsException you would instead just re-throw the one that you caught since it is still the same error you just did some handling and now are passing it on in case others need to also handle it.  So the code would be:
throw e;  //That is the name you gave the exception you caught